### DISCUSSION TITLE
@alwatr/validator

### DISCUSSION THREAD

**Discussion by asmanp2012** [AliMD/alwatr#116] [Ideas]
We need a package to validate input and string.
What is our policy on this issue? `Create a new package for that` or `Use the existing javascript package`.

1- [validator](https://github.com/validatorjs/validator.js)
2- [ajv](https://github.com/ajv-validator/ajv)
3- [joi](https://github.com/sideway/joi)
4- [yup](https://github.com/jquense/yup)

**Comment by alimd** [comment]
WOW to ajv!!

**Reply by alimd** [reply]
منو کلا با یک مفهوم جدید و به شدت کاربردی آشنا کردی دمت گرم

**Comment by alimd** [comment]
https://github.com/korzio/djv

**Comment by alimd** [comment]
پدرم در اومد تا اینو پیداش کردم چون اسم بی ربطی داره هر دفعه میخوام کلی باید دنبالش بگردم
واقعا برای اینکار بهترینه و تو ذهنم همچین چیزی بود

https://nosir.github.io/cleave.js/

**Comment by asmanp2012** [comment]
**`validator.js` doesn't check the type of input.**

**Comment by MM25Zamanian** [comment]
درک نمیکنم وقتی جی سان ها از سمت سرور میان که کاملا استیبل هست چرا باید ولیدیت بشن؟؟ 

@asmanp2012

**Reply by asmanp2012** [reply]
کلا json وقتی از یک منبع خارجی میادش غیر قابل اعتماد هستش گر چه تو ui کمی سعی میکنیم اعتماد کنیم به سرور ولی 
از سمت کلاینت وقتی چیزی رو به سمت سرور میفرستیم کلا غیر قابل اعتماده

**Reply by njfamirm** [reply]
> کلا json وقتی از یک منبع خارجی میادش غیر قابل اعتماد هستش گر چه تو ui کمی سعی میکنیم اعتماد کنیم به سرور ولی از سمت کلاینت وقتی چیزی رو به سمت سرور میفرستیم کلا غیر قابل اعتماده

کلی وقت گذاشتم روی ajv برای چک کردن دیتایی که از سرور میاد فرانت
بعد خسته شدم دوباره جملتون رو خوندم دیدم گفتین لازم نیست 😭😭 @asmanp2012
